assets/javascripts/discourse/controllers/admin-plugins-explorer.js.es6

```diff
@@ -117,16 +124,64 @@ export default Ember.Controller.extend({
       .finally(() => this.set("loading", false));
   },
 
+  async _importQuery(file) {
+    const json = await this._readFileAsTextAsync(file);
+    const query = await this._parseQuery(json);
+    const record = this.store.createRecord("query", query);
+    const response = await record.save();
+    return response.target;
+  },
+
+  async _parseQuery(json) {
+    const parsed = JSON.parse(json);
+    const query = parsed.query;
+    if (!query || !query.sql) {
+      throw new TypeError();
+    }
+    query.id = 0; // 0 means no Id yet
+    return query;
+  },
+
+  _readFileAsTextAsync(file) {
```

Comment:
Shouldn't this function be marked `async` if you are `await`ing it in `_importQuery`?

Reply:
No, we don't need it here. Normally, you need `async` if you're awaiting something _inside_ the function. This function just create and return a promise, and we're awaiting that promise outside.

Reply:
Ah cool I didn't realise that thanks for clarifying